wd_core.au3

```diff
@@ -1353,7 +1334,10 @@ Func __WD_Get($sURL)
 		$iResult = $_WD_ERROR_InvalidValue
 	EndIf
 
-	Local $sMessage = (($sResponseText) ? ("ResponseText=" & StringLeft($sResponseText, $_WD_RESPONSE_TRIM) & "...") : (""))
+	__WD_ConsoleWrite($sFuncName & ": URL=" & $sURL, $_WD_DEBUG_Full)
+	Local $sMessage = "HTTP status = " & $_WD_HTTPRESULT
+	$sMessage &= ($_WD_DEBUG = $_WD_DEBUG_Info And $iResult) ? (" ResponseText=" & StringLeft($sResponseText, $_WD_RESPONSE_TRIM) & "...") : ("") ; in case of $_WD_DEBUG_Info  >  trimmed $sResponseText but only when @error is set
+	$sMessage &= ($_WD_DEBUG = $_WD_DEBUG_Full) ? (" ResponseText=" & $sResponseText) : ("") ; in case of $_WD_DEBUG_Full  >  Full $sResponseText
```

Comment:
I think this could be written so that it is less confusing. I can do it if you would like.

Reply:
Sure. Don't hold back.

Reply:
Take a look here: [__WD_Get $sMessage refactoring](https://github.com/Danp2/au3WebDriver/pull/290/commits/aaca3eb036ac41bb4a45ff1c922e5107c330d10b)

Reply:
Do you want from me to do the same with `_WD_Post/Delete`

Reply:
I like the direction this is going. My concern at this point is with the added checking of `$iResult` and logging the response text only when an error has occurred. While this would eliminate the "clutter" from the log, I believe it could hamper debugging efforts by excluding details that could be useful. 

IMO, the user should use `_WD_DEBUG_Error` if they only want to see error related content.

Reply:
We need somehow to use `$_WD_RESPONSE_TRIM`

in case of `$_WD_DEBUG_Info` trimmed $sResponseText is put to the log.
in case of `$_WD_DEBUG_Full` Full $sResponseText is put to the log.

from my point of view 
`$_WD_DEBUG_Info` is something between `$_WD_DEBUG_error` and `$_WD_DEBUG_Full`
What this mean all `$_WD_DEBUG_Info` related messages should be put to log only when `@error` was fired.

Reply:
What do you think about it?

Reply:
> We need somehow to use $_WD_RESPONSE_TRIM

I think it's fine to not trim the response when `$_WD_DEBUG_Full` is in use

> From my point of view $_WD_DEBUG_Info is like a extended type of $_WD_DEBUG_error

No, that wasn't the intention. As you may recall, `$_WD_DEBUG_Full` didn't always exist, so `$_WD_DEBUG_Info` was the highest debugging level at one time. 

I believe that we previously discussed only logging the results of __WD_Get/Put/Delete when `$_WD_DEBUG_Full` is in use. That would reduce some of the duplication that occurs when `$_WD_DEBUG_Info` is in use

Reply:
So what about removing `$_WD_RESPONSE_TRIM` ?

Reply:
Should `$_WD_DEBUG_error` shows descriptions for errors or only raw numbers ?
If `$_WD_DEBUG_error`  should show only raw numbers then `$_WD_DEBUG_info`  can also show descriptions declared in:
```autoit
Global Const $aWD_ERROR_DESC[$_WD_ERROR_COUNTER]
```

Reply:
> So what about removing $_WD_RESPONSE_TRIM ?

What about it? Did you see where I said the following?

> > I think it's fine to not trim the response when $_WD_DEBUG_Full is in use

I'm confused by you asking about it again.

Reply:
Sorry for cofusion.
I should have said it more precisely.

This is how I understand it now:

Only `$_WD_DEBUG_Full` should put HTTP response to log , and it can't be trimmed.
This means that ResponseText is never to be passed to the console in a truncated version, but only in the full version provided it is used with `$_WD_DEBUG_Full` .

Hence the `$_WD_RESPONSE_TRIM` feature should be removed from the UDF and this should be treated as `Script Breaking Change`

Does my reasoning align with yours?

Reply:
> Should $_WD_DEBUG_error shows descriptions for errors or only raw numbers ?

`__WD_Error` currently shows both the string description and the error number, correct? Unless I'm missing something, then I wouldn't expect this to change based upon $_WD_DEBUG 

> $_WD_DEBUG_info should also show parameters passed to the function 

I believe this is outside the scope of this PR and should be evaluated elsewhere

Reply:
> > Should $_WD_DEBUG_error shows descriptions for errors or only raw numbers ?
> 
> `__WD_Error` currently shows both the string description and the error number, correct? Unless I'm missing something, then I wouldn't expect this to change based upon $_WD_DEBUG
> 
> > $_WD_DEBUG_info should also show parameters passed to the function
> 
> I believe this is outside the scope of this PR and should be evaluated elsewhere

ok

Reply:
> Do you want from me to do the same with `_WD_Post/Delete` ?

This question is still valid.

It was related to:
> Take a look here: [__WD_Get $sMessage refactoring](https://github.com/Danp2/au3WebDriver/pull/290/commits/aaca3eb036ac41bb4a45ff1c922e5107c330d10b)

Reply:
> Do you want from me to do the same with _WD_Post/Delete ?

Yes, but not until we finish with the changes to __WD_Get. Once that is finalized, then the changes can be carried over to __WD_Post/Delete.

> Does my reasoning align with yours?

No, I don't think so. I'll have to think about this some more before responding further.

Reply:
> My concern at this point is with the added checking of $iResult and logging the response text only when an error has occurred. 

This still needs to be resolved.

Reply:
> So what about removing $_WD_RESPONSE_TRIM ?

I now understand what you were asking here. However, I believe that we should continue to support `$_WD_RESPONSE_TRIM` with `$_WD_DEBUG_Full` since some users may not want to log the complete response in all cases.

Maybe change the default value of `$_WD_RESPONSE_TRIM` to -1, which would indicate that no trimming should occur?

Reply:
Ok

Reply:
Please take a look on:
[__WD_Get ResponseText $_WD_RESPONSE_TRIM](https://github.com/Danp2/au3WebDriver/pull/290/commits/d1a8cb3752018def122be5e7c7de516a72aa2b0f)

Reply:
> > Do you want from me to do the same with _WD_Post/Delete ?
> 
> Yes, but not until we finish with the changes to __WD_Get. Once that is finalized, then the changes can be carried over to __WD_Post/Delete.

I just added 2 new review to not forget about this things to do.

Reply:
> > My concern at this point is with the added checking of $iResult and logging the response text only when an error has occurred.
> 
> This still needs to be resolved.

Please take a look on:
[__WD_Get ResponseText $_WD_RESPONSE_TRIM](https://github.com/Danp2/au3WebDriver/pull/290/commits/d1a8cb3752018def122be5e7c7de516a72aa2b0f)

Reply:
> However, I believe that we should continue to support `$_WD_RESPONSE_TRIM` with `$_WD_DEBUG_Full` since some users may not want to log the complete response in all cases.

As I understand it, `$_WD_RESPONSE_TRIM` should not be used with `$_WD_DEBUG_Info` 
For this reason I already did the change in my recent commit:  [__WD_Get ResponseText $_WD_RESPONSE_TRIM](https://github.com/Danp2/au3WebDriver/pull/290/commits/d1a8cb3752018def122be5e7c7de516a72aa2b0f).

Reply:
> Maybe change the default value of `$_WD_RESPONSE_TRIM` to -1, which would indicate that no trimming should occur?

Take a look on:
[Global $_WD_RESPONSE_TRIM = -1](https://github.com/Danp2/au3WebDriver/pull/290/commits/4e401aa710f481dfa931cc63211dbe3acda673dc)

Reply:
btw. 
Please review: `; Trim response string to given value for debug output`
I think that this comment is not accurate any more.

Reply:
So whats the status here ?
Can we go further with this what we got or is here (this conversation) something to change ?

Reply:
If you have some Ideas to this conversation part, please create commit.